src/components/VmActions/index.js

```diff
@@ -89,50 +92,105 @@ EmptyAction.propTypes = {
  * Active actions on a single VM-card.
  * List of actions depends on the VM state.
  */
-const VmActions = ({ vm, isOnCard = false, onGetConsole, onShutdown, onRestart, onStart, onSuspend }) => {
-  const status = vm.get('status')
-
-  const confirmShutdown = {
-    confirmText: 'Shut down?',
-    cancelText: 'Cancel',
+class VmActions extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      openModal: false,
+    }
+    this.onConsoleClick = this.onConsoleClick.bind(this)
+    this.onModalClose = this.onModalClose.bind(this)
   }
 
-  const confirmRestart = {
-    confirmText: 'Restart?',
-    cancelText: 'Cancel',
+  onConsoleClick () {
+    this.setState({
+      openModal: true,
+    })
+    this.props.checkConsole()
   }
 
-  const confirmSuspend = {
-    confirmText: 'Suspend?',
-    cancelText: 'Cancel',
+  onModalClose () {
+    this.setState({
+      openModal: false,
+    })
+    this.props.modalClose()
   }
-  let consoleProtocol = ''
-  if (!vm.get('consoles').isEmpty()) {
-    const vConsole = vm.get('consoles').find(c => c.get('protocol') === 'spice') ||
-      vm.getIn(['consoles', 0])
-    const protocol = vConsole.get('protocol').toUpperCase()
-    consoleProtocol = `Open ${protocol} Console`
+
+  render () {
+    let {
+      vm,
+      VmAction,
+      isOnCard = false,
+      onGetConsole,
+      onShutdown,
+      onRestart,
+      onStart,
+      onSuspend,
+    } = this.props
+    const status = vm.get('status')
+
+    const confirmShutdown = {
+      confirmText: 'Shut down?',
+      cancelText: 'Cancel',
+    }
+
+    const confirmRestart = {
+      confirmText: 'Restart?',
+      cancelText: 'Cancel',
+    }
+
+    const confirmSuspend = {
+      confirmText: 'Suspend?',
+      cancelText: 'Cancel',
+    }
+    let consoleProtocol = ''
+    if (!vm.get('consoles').isEmpty()) {
+      const vConsole = vm.get('consoles').find(c => c.get('protocol') === 'spice') ||
+        vm.getIn(['consoles', 0])
+      const protocol = vConsole.get('protocol').toUpperCase()
+      consoleProtocol = `Open ${protocol} Console`
+    }
+
+    if (vm.get('sessions') &&
+      !(vm.get('sessions').filter(c => c.get('consoleUser')).isEmpty())) {
```

Comment:
It's correct, the existence of potentially existing parallel console session is checked right before the console is downloaded (the new VmActions/sagas.js).

Just an idea:
Within this saga, the consoleUser might be set to some constant pessimistically enforcing the state the console is in use and once the fetchVmSessions() finishes, correct value is set